tests/02_eradiate/01_unit/scenes/shapes/test_buffermesh.py

```diff
@@ -0,0 +1,25 @@
+import os
+import tempfile
+
+import pytest
+
+from eradiate.contexts import KernelDictContext
+from eradiate.scenes.shapes import BufferMeshShape
+
+
+def test_construct(modes_all_double):
+    ctx = KernelDictContext()
+    mesh = BufferMeshShape(
+        vertices=[[1, 0, 0], [0, 1, 0], [0, 0, 1]], faces=[[1, 2, 3]]
+    )
+    assert mesh.kernel_dict(ctx=ctx)
+
+    # flat list of vertex pos fails
+    with pytest.raises(IndexError):
```

Comment:
An incorrect shape signaled by `IndexError` is inappropriate. You should add a validator step to check for this and raise an exception (probably `ValueError` as usual) with a meaningful error message. IIRC this kind of behaviour is implemented in other components (maybe the abstract canopies?).